### PR TITLE
Add 3.0.1-RC1 blog article

### PR DESCRIPTION
**Pull request by anatoliykmetyuk**
[skip ci]

### CODE REVIEW

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+Having said that, we still encourage people to play with the experimental features from the `NIGHTLY` compier versions and discuss their findings. Without the curious and adventurous part of the community playing with the new features, there is no way of knowing what they are good for, and no way to decide whether they should be dropped or promoted to a stable feature.
+
+More about this change you can read in the PR [#12102](https://github.com/lampepfl/dotty/pull/12102).
+
+# Kind-projector work
+This release also brings extra features for the [Kind Projector](https://docs.scala-lang.org/scala3/guides/migration/plugin-kind-projector.html) migration support. First, PR [#12378](https://github.com/lampepfl/dotty/pull/12378) allows `_` as type lambda placeholder. Second, PR [#12341](https://github.com/lampepfl/dotty/pull/12341) brings support for the variance annotations on the placeholder. This work enhances the ability to cross-compile Scala 2 code that uses the Kind Projector plugin to Scala 3.
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:27]: /cc @neko-kai

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  |                  failed since selector  B
+  |                  does not match  case A => Int
+  |                  and cannot be shown to be disjoint from it either.
+  |                  Therefore, reduction cannot advance to the remaining case
+  |
+  |                    case B => String
+```
+
+# Scaladoc
+We have updated the [documentation](http://dotty.epfl.ch/docs/usage/scaladoc/index.html) for Scaladoc making it easier for you to get started. Also, PR [#11582](https://github.com/lampepfl/dotty/pull/11582) has added the snippet compiler to ensure the snippets in your scaladoc documentation comments aren't broken. You can read more about this feature on the [mailing list](https://contributors.scala-lang.org/t/snippet-validation-in-scaladoc-for-scala-3/4976).
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:61]: /cc @pikinier20 @romanowski

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+# Metaprogramming
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:63]: /cc @nicolasstucki

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+author: Anatolii Kmetiuk
+authorImg: /images/anatolii.png
+date: 2021-06-07
+---
+
+Hello! We are happy to announce Scala 3.0.1-RC1 – the first release candidate in the post-3.0.0 era. With this release, we continue the work on making the compiler even more stable.
+
+<!--more-->
+
+# Experimental language features policy
```
Comment (anatoliykmetyuk) [inline docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:13]: /cc @odersky @sjrd @smarter

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+# Metaprogramming
+A lot of metaprogramming work was focused on improving the performance. Some of the notable PRs include:
+
+- Cache quote unpickling [#12242](https://github.com/lampepfl/dotty/pull/12242)
+- Avoid pickled tasty for some captured quote reference [#12248](https://github.com/lampepfl/dotty/pull/12248)
+- Improve quote matcher performance [#12418](https://github.com/lampepfl/dotty/pull/12418)
+- Port scala.quoted.runtime.impl.QuoteMatcher [#12402](https://github.com/lampepfl/dotty/pull/12402)
+
+A way to abort macro expansion was enhanced in PR [#12056](https://github.com/lampepfl/dotty/pull/12056). Now you can use `errorAndAbort(msg: String)` to stop macro expansion with an error, and `expr.valueOrAbort` to abort macro expansion if a given expression doesn't represent a value or possibly contains side effects.
```
Comment (nicolasstucki) [inline docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:71]:
```suggestion
```

This is experimental. We should not mention it here as users will no be able to use it.

File: docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md (hunk 1 above)
Comment (neko-kai) [reply docs/blog/_posts/2021-06-07-scala3.0.1-rc1-release.md:27]: @anatoliykmetyuk I would add a link to the migration guide which has more details on how to to make use of it from a user perspective - https://docs.scala-lang.org/scala3/guides/migration/plugin-kind-projector.html – and a quick mention that `-Ykind-projector:underscores` option is required to enable it.

**Pull request merged**